tests/test_wait_signal.py

```diff
@@ -640,6 +641,332 @@ def test_signals_and_callbacks_length_mismatch(self, qtbot, signaller):
+
+        with qtbot.waitSignal(signal=signaller.signal_args, timeout=200, check_params_cb=cb, raising=False) as blocker:
+            signaller.signal_args.emit('2', 2)
+            signaller.signal_args.emit('1', 1)
+        assert blocker.all_args == [('2', 2), ('1', 1)]
+
+
+def get_mixed_signals_with_guaranteed_name(signaller):
+    """
+    Returns a list of signals with the guarantee that the signals have names (i.e. the names are
+    manually provided in case of using PySide, where the signal names cannot be determined at run-time).
+    """
+    if qt_api.pytest_qt_api == 'pyside':
+        signals = [(signaller.signal, "signal()"), (signaller.signal_args, "signal_args(QString,int)"),
+                   (signaller.signal_args, "signal_args(QString,int)")]
+    else:
+        signals = [signaller.signal, signaller.signal_args, signaller.signal_args]
+    return signals
+
+
+class TestAllSignalsAndArgs:
+    """
+    Tests blocker.all_signals_and_args (waitSignals() blocker) is filled with the namedtuple SignalAndArgs for each
+    received expected signal (irrespective of the order parameter).
+    """
+
+    def test_empty_when_no_signal(self, qtbot, signaller):
+        """Tests that all_signals_and_args is empty when no expected signal is emitted."""
+        signals = get_mixed_signals_with_guaranteed_name(signaller)
+        with qtbot.waitSignals(signals=signals, timeout=200, check_params_cbs=None, order="none",
+                               raising=False) as blocker:
+            pass
+        assert blocker.all_signals_and_args == []
+
+    def test_empty_when_no_signal_name_available(self, qtbot, signaller):
+        """
+        Tests that all_signals_and_args is empty even though expected signals are emitted, but signal names aren't
+        available.
+        """
+        if qt_api.pytest_qt_api != 'pyside':
+            pytest.skip("test only makes sense for PySide, whose signals don't contain a name!")
+
+        with qtbot.waitSignals(signals=[signaller.signal, signaller.signal_args, signaller.signal_args],
+                               timeout=200, check_params_cbs=None, order="none", raising=False) as blocker:
+            signaller.signal.emit()
+            signaller.signal_args.emit('1', 1)
+        assert blocker.all_signals_and_args == []
+
+    def test_non_empty_on_timeout_no_cb(self, qtbot, signaller):
+        """
+        Tests that all_signals_and_args contains the emitted signals. No callbacks for arg-evaluation are provided. The
+        signals are emitted out of order, causing a timeout.
+        """
+        signals = get_mixed_signals_with_guaranteed_name(signaller)
+        with qtbot.waitSignals(signals=signals, timeout=200, check_params_cbs=None, order="simple",
+                               raising=False) as blocker:
+            signaller.signal_args.emit('1', 1)
+            signaller.signal.emit()
+        assert blocker.signal_triggered is False
```

Comment:
just use `assert not blocker.signal_triggered` (same for other places where you use `is True` and `is False`)